### ISSUE TITLE
[Bug]: mocking does not work

### ISSUE THREAD

**Issue by UebelAndre** [aspect-build/rules_jest#123]
### What happened?

I have a repro here: https://github.com/UebelAndre/rules_jest_example

The README shows a test that mocks an object in another module working as expected outside of Bazel but when using Bazel with `rules_jest`, `rules_ts`, and `rules_js`, the test fails and the object is never mocked.

```
INFO: Analyzed 14 targets (0 packages loaded, 0 targets configured).
INFO: Found 13 targets and 1 test target...
FAIL: //foo:foo_tests (see /private/var/tmp/_bazel_user/e187ae0f02551085cd56d499814ad066/execroot/rules_jest_example/bazel-out/darwin_arm64-fastbuild/testlogs/foo/foo_tests/test.log)
INFO: From Testing //foo:foo_tests:
==================== Test output for //foo:foo_tests:
 FAIL  src/index.test.js
  ✕ show point was mocked (2 ms)

  ● show point was mocked

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: -1

      at Object.<anonymous> (src/index.test.ts:8:21)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.211 s
Ran all test suites.
================================================================================
INFO: Elapsed time: 1.989s, Critical Path: 1.83s
INFO: 4 processes: 1 internal, 2 darwin-sandbox, 1 local.
INFO: Build completed, 1 test FAILED, 4 total actions
//foo:foo_tests                                                          FAILED in 0.6s
  /private/var/tmp/_bazel_user/e187ae0f02551085cd56d499814ad066/execroot/rules_jest_example/bazel-out/darwin_arm64-fastbuild/testlogs/foo/foo_tests/test.log

Executed 1 out of 1 test: 1 fails locally.
```

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: bazel 6.1.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
rules_js-1.24.1
rules_ts-1.3.2
rules_jest-0.18.1

Language(s) and/or frameworks involved:

### How to reproduce

```shell
See https://github.com/UebelAndre/rules_jest_example
```

**Comment by jbedard** [comment]
I assume this is not typescript related but would you be able to verify that my removing rules_ts from the repro? Simplifying the repro would also make it easier to start investigate 🙏

**Comment by UebelAndre** [comment]
My javascript is fairly weak so I don't know how exactly to setup the same repro with just javascript. This was the most simplified I could make it 😅

**Comment by jbedard** [comment]
Actually it seems to work fine if you drop `ts-jest`. I assume the bazel compiled ts and `ts-jest` compiled version are conflicting? 🤷

**Comment by UebelAndre** [comment]
How is jest getting ts to compile? I thought it'd only be working with javascript?

**Comment by jbedard** [comment]
You're using the `ts-jest` jest transformer. I'm not too sure how it works when you're loading .js files but that seems to interfere with mocking.

Generally when using bazel you should compile ts once with `ts_project` and everything else (such as jest) should consume the output of that. That ts compilation can then be cached and things like jest, bundlers etc shouldn't be compiling it again...

**Comment by jbedard** [comment]
Closing since it seems related to mixing `ts-jest` and `ts_project` but feel free to continue the conversation here.
